engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -1058,33 +1018,97 @@ verifyEscape = do
-guessAlter _ (TriggerTile{ttfeature=TK.OpenTo _} : _) _ = "cannot be opened"
-guessAlter COps{cotile} (TriggerTile{ttfeature=TK.CloseTo _} : _) t
-  | Tile.isOpenable cotile t = "already closed"
-guessAlter _ (TriggerTile{ttfeature=TK.CloseTo _} : _) _ = "cannot be closed"
-guessAlter _ _ _ = "never mind"
-
 -- * AlterWithPointer
 
 -- | Try to alter a tile using a feature under the pointer.
 alterWithPointerHuman :: MonadClientUI m
-                      => [TriggerTile] -> m (FailOrCmd RequestTimed)
-alterWithPointerHuman ts = do
-  COps{corule=RuleContent{rXmax, rYmax}, cotile} <- getsState scops
-  lidV <- viewedLevelUI
+                      => m (FailOrCmd RequestTimed)
+alterWithPointerHuman = do
+  COps{corule=RuleContent{rXmax, rYmax}} <- getsState scops
   -- Not @ScreenContent@, because not drawing here.
-  lvl <- getLevel lidV
   K.PointUI x y <- getsSession spointer
   let (px, py) = (x `div` 2, y - K.mapStartY)
       tpos = Point px py
-      t = lvl `at` tpos
   if px >= 0 && py >= 0 && px < rXmax && py < rYmax
-  then alterTileAtPos ts tpos $ "the" <+> TK.tname (okind cotile t)
-  else failWith "never mind"
+  then alterTileAtPos (Just tpos)
+  else alterTileAtPos Nothing
+
+-- * CloseDir 
+
+-- | Close nearby open tile; ask for direction, if there is more than one.
+closeDirHuman :: MonadClientUI m
+              => m (FailOrCmd RequestTimed)
+closeDirHuman = do
+  COps{cotile} <- getsState scops
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  lvl <- getLevel $ blid b
+  let vPts = vicinityUnsafe $ bpos b 
+      openPts = filter (Tile.isClosable cotile . at lvl) vPts
+  case openPts of 
+    []  -> failSer CloseNothing
+    [o] -> closeTileAtPos $ Just o
+    _   -> pickPoint "close" >>= closeTileAtPos 
+   
+-- | Close tile at given position.
+closeTileAtPos :: MonadClientUI m
+               => Maybe Point -> m (FailOrCmd RequestTimed)
+closeTileAtPos mp = case mp of
+  Nothing -> failWith "never mind"
+  Just p -> do
+    COps{cotile} <- getsState scops
+    leader <- getLeaderUI
+    b <- getsState $ getActorBody leader
+    if (p `chessDist` bpos b == 1) then do
+      actorSk <- leaderSkillsClientUI
+      lvl <- getLevel $ blid b
+      let alterSkill = Ability.getSk Ability.SkAlter actorSk
+      if | Tile.isOpenable cotile $ lvl `at` p 
+          -> failSer CloseClosed
+         | alterSkill <= 1
+          -> failSer AlterUnskilled
+         | otherwise 
+          -> do
+             msgAddDone p "close"
+             return $ Right $ ReqAlter p 
+    else failSer CloseDistant 
+
+-- | Adds message with proper names.
+msgAddDone :: MonadClientUI m
+           => Point -> Text
+           -> m ()
+msgAddDone p verb = do
+  COps{cotile} <- getsState scops
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  lvl <- getLevel $ blid b
+  let s = case (T.words . TK.tname $ okind cotile $ lvl `at` p) of
+          [] -> "thing"
+          [x] -> x
+          (_ : xs) -> T.unwords xs
```

Comment:
Sadly, this has to be a bit more complex: when closing we can assume "open window", but when altering, we can have "rubble pile", which we remove, causing some effects and perhaps discovering treasure. Just "pile" won't sound good. Similarly with switches we toggle, etc. This is barely used, but it will be and also, this is a game enigine, so we are allowed to be general. Hmm, actually, for KISS, I might even like "you close the open gate north", but I leave it for you to decide.